### PR TITLE
DEVPROD-34496 Track S3 costs for test log uploads

### PR DESCRIPTION
**Pull request by malikchaya2**
DEVPROD-34496

### Description

Test logs (not to be confused with test results that are uploaded to a different bucket) are uploaded to the same S3 bucket as task/agent/system logs, but their PUT and storage costs were excluded from the per-task S3 cost calculation. This adds test_log as a tracked log type so test log uploads contribute to cost reporting alongside the other log types.

### Testing
Added tests

### CODE REVIEW

File: model/task/evergreen_sender.go [hunk 1]
```diff
@@ -255,5 +257,15 @@ func (s *evergreenSender) flush(ctx context.Context) error {
 	s.bufferSize = 0
 	s.lastFlush = time.Now()
 
+	if s.opts.S3Usage != nil && puts > 0 {
```
Comment (maddaak) [inline model/task/evergreen_sender.go:253]: We can have puts > 0 even then there error. We need to move this back to line 247 along with the mutex hold

File: agent/internal/taskoutput/test_log.go [hunk 2]
```diff
@@ -29,9 +30,11 @@ import (
 var defaultTestLogSequenceSize = int64(1e7)
 
 // AppendTestLog appends log lines to the specified test log for the given task
-// run.
-func AppendTestLog(ctx context.Context, tsk *task.Task, redactionOpts redactor.RedactionOptions, testLog *testlog.TestLog) error {
-	sender, err := task.NewTestLogSender(ctx, *tsk, task.EvergreenSenderOptions{}, testLog.Name, 0)
+// run. s3UsageMu, when non-nil, is held while incrementing S3 usage metrics;
+// callers that invoke AppendTestLog concurrently with a shared s3UsageTracker
+// must supply the same mutex across all concurrent calls.
+func AppendTestLog(ctx context.Context, tsk *task.Task, redactionOpts redactor.RedactionOptions, testLog *testlog.TestLog, s3UsageTracker *s3usage.S3Usage, s3UsageMu *sync.Mutex) error {
```
Comment (maddaak) [inline agent/internal/taskoutput/test_log.go]: NIT: `s3UsageTracker *s3usage.S3Usage` --> `s3Usage *s3usage.S3Usage`

File: agent/internal/taskoutput/test_log_test.go [hunk 3]
```diff
@@ -76,7 +76,7 @@ func TestAppendTestLog(t *testing.T) {
 			tsk.TaskOutputInfo.TestLogs.BucketConfig.Name = t.TempDir()
 			testLog.Lines = testCase.input
 
-			require.NoError(t, AppendTestLog(ctx, tsk, testCase.redactOpts, testLog))
+			require.NoError(t, AppendTestLog(ctx, tsk, testCase.redactOpts, testLog, nil, nil))
```
Comment (maddaak) [inline agent/internal/taskoutput/test_log_test.go]: NIT: could we add a test for non nil s3usage and mutex

**Review by maddaak — CHANGES_REQUESTED** [review]

**Review by maddaak — APPROVED** [review]
